trieste/acquisition/rule.py

```diff
@@ -2073,6 +2073,191 @@ def update(
+    this trust region is not suitable for problems with a large number of points.
+    """
+
+    def __init__(
+        self,
+        global_search_space: DiscreteSearchSpace,
+        beta: float = 0.7,
+        kappa: float = 1e-4,
+        min_eps: float = 1e-2,
+        region_index: Optional[int] = None,
+        input_active_dims: Optional[Union[slice, Sequence[int]]] = None,
+    ):
+        """
+        Select a random initial location from the global search space and select the initial
+        neighbors within the trust region.
+
+        :param global_search_space: The global search space this search space lives in.
+        :param beta: The inverse of the trust region contraction factor.
+        :param kappa: Scales the threshold for the minimal improvement required for a step to be
+            considered a success.
+        :param min_eps: The minimal size of the search space. If the size of the search space is
+            smaller than this, the search space is reinitialized.
+        :param region_index: The index of the region in a multi-region search space. This is used to
+            identify the local models and datasets to use for acquisition. If `None`, the
+            global models and datasets are used.
+        :param input_active_dims: The active dimensions of the input space, either a slice or list
+            of indices into the columns of the space. If `None`, all dimensions are active.
+        """
+        super().__init__(global_search_space, region_index, input_active_dims)
+        self._beta = beta
+        self._kappa = kappa
+        self._min_eps = min_eps
+        self._initialized = False
+        self._step_is_success = False
+        self._init_location()
+        self._compute_global_distances()
+        self._init_eps()
+        self._update_neighbors()
+        self._y_min = np.inf
+
+    @property
+    def location(self) -> TensorType:
+        """Center point of the region."""
+        return tf.reshape(tf.gather(self.global_search_space.points, self._location_ix), (-1,))
+
+    def _init_location(self) -> None:
+        # Random initial location index from the global search space.
+        self._location_ix = tf.random.categorical(
+            tf.ones(
+                (
+                    1,
+                    self.global_search_space.points.shape[0],
+                )
+            ),
+            1,
+        )[0]
+
+    def _init_eps(self) -> None:
+        global_lower = self.global_search_space.lower
+        global_upper = self.global_search_space.upper
+        self.eps = 0.5 * (global_upper - global_lower) / (5.0 ** (1.0 / global_lower.shape[-1]))
+
+    def _compute_global_distances(self) -> None:
+        # Pairwise distances along each axis in the global search space.
+        points = self.global_search_space.points
+        self._global_distances = tf.abs(
+            tf.expand_dims(points, -2) - tf.expand_dims(points, -3)
+        )  # [num_points, num_points, D]
```

Comment:
Yes it is symmetric. I did think about this, and decided it isn't worth the complexity of keeping only part of the matrix. You could in principle use a `tf.sparse.SparseTensor`, but that seems unnecessary since we expect the number of points to be small. Also, that might have a performance impact due to the operations being less likely to be parallelizable on a GPU. Note: in gpflow we also have symmetric covariance matrices, but we choose to always keep the full matrices.

Reply:
ok, if the benefits are not clear, lets stick to this simpler implementation